core/src/App.html

```diff
@@ -42,45 +71,98 @@
             navigateTo(matchedPath);
         }
     };
 
+    const sendContextToClient = (component, config, callbackFn) => {
+        const goBackValue = isFunction(callbackFn) ? callbackFn() : callbackFn;
+        config.iframe.contentWindow.postMessage({
+            msg: 'luigi.init',
+            context: JSON.stringify(Object.assign({}, component.get().context, goBackValue)),
+            internal: JSON.stringify(component.prepareInternalData())
+        }, '*');
+    };
+
     export default {
+        data() {
+            return {
+                preservedViews: [
+                // {
+                //     path: '/project/p2/settings',
+                //     nextPath: '/project/p2',
+                //     context: {}
+                //     iframe: Element
+                // }
+                ]
+            }
+        },
         oncreate() { },
         onstate({ changed, current, previous }) { },
         actions: {
             init(node) {
                 const config = {
                     iframe: null,
                     navigateOk: null,
-                    builderCompatibilityMode: window['builderCompatibilityMode'] || false
+                    builderCompatibilityMode: window.builderCompatibilityMode || false
                 };
 
                 window.addEventListener("message", (e) => {
                     if ('luigi.get-context' === e.data.msg && config.iframe) {
-                        config.iframe.contentWindow.postMessage({ msg: 'luigi.init', context: JSON.stringify(this.get().context) }, '*');
+                        sendContextToClient(this, config, {});
                     }
 
                     if ('luigi.navigate.ok' === e.data.msg) {
                         config.navigateOk = true;
                     }
 
                     if ('luigi.navigation.open' === e.data.msg) {
-                        handleNavigation(this, e.data);
+                        console.log('luigi.navigation.open', e.data);
+                        handleNavigation(this, e.data, config, node);
+                    }
+
+                    if ('luigi.navigation.back' === e.data.msg) {
+                        // go back: context from the view
+                        const pvs = this.get().preservedViews;
+                        if (pvs && pvs.length) {
+                            // remove current active iframe and data
+                            setPreviousActiveIframe(node);
+                            const pv = pvs.pop();
+                            // set new active iframe and preservedViews
+                            config.iframe = getActiveIframe(node);
+                            this.set({'preservedViews': pvs});
+                            // TODO: check if handleNavigtion or history pop to update hash / path
+                            handleNavigation(this, {params: {link: pv.path}}, config, node);
+
+                            // send stored context
```

Comment:
Could you please explain why this code is commented out? We are not giving the option to update the context when we navigate back to an existing node